client/client-multi/client-objective-c/Kaa/DefaultBootstrapTransport.m

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "DefaultBootstrapTransport.h"
+#import "KaaLogging.h"
+
+#define TAG @"DefaultBootstrapTransport >>>"
+
+@interface DefaultBootstrapTransport ()
+
+@property (nonatomic,strong) id<BootstrapManager> manager;
+@property (nonatomic,strong) NSString *sdkToken;
+@property (atomic) int increment;
+
+@end
+
+@implementation DefaultBootstrapTransport
+
+- (instancetype)initWithToken:(NSString *)sdkToken {
+    self = [super init];
+    if (self) {
+        self.sdkToken = sdkToken;
+        self.increment = 0;
+    }
+    return self;
+}
+
+- (SyncRequest *)createResolveRequest {
+    if (!self.clientState) {
+        return nil;
+    }
+    SyncRequest *request = [[SyncRequest alloc] init];
```

Comment:
It is a bit strange that a bootstrap transport is responsible for creating the whole client sync.